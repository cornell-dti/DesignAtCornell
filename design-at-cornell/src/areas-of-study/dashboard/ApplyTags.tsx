import React from 'react';
import { ApplyTagsContainer } from '../AreasOfStudyStyles';

const ApplyTags = () => {
  const designAreaTags = [
    'Art + Design',
    'Digital Design',
    'Industrial Design',
    'Policy Design',
    'Spatial Design',
  ];
  const schools = [
    'AAP',
    'Agriculture and Life Sciences',
    'Arts and Sciences',
    'Engineering',
    'Human Ecology',
    'ILR',
    'SC Johnson School of Business',
  ];

  const checkboxes = (tags: string[]) => (
    <form>
      {tags.map((tag) => (
        <div>
          <input key={tag} type="checkbox" id={tag} />
          <label htmlFor={tag}>{tag}</label>
        </div>
      ))}
    </form>
  );

  return (
    <ApplyTagsContainer>
      <h5>Apply Tags</h5>
      <h6>Design Areas</h6>
      {checkboxes(designAreaTags)}
      <h6>Schools</h6>
      {checkboxes(schools)}
    </ApplyTagsContainer>
  );
};

export default ApplyTags;
