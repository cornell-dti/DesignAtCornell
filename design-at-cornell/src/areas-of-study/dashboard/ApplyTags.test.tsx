import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ApplyTags from './ApplyTags';

describe('<ApplyTags />', () => {
  it('renders correctly', () => {
    const snapshot = renderer.create(<ApplyTags />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
