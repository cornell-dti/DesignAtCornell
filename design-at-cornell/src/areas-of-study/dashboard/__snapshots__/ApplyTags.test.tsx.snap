// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplyTags /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 20%;
  height: auto;
  background: rgba(142,214,99,0.17);
  padding: 60px;
}

.c0 h5 {
  font-weight: bold;
  font-size: 20px;
  line-height: 70px;
  color: black;
  margin: 0;
}

.c0 h6 {
  font-weight: bold;
  font-size: 16px;
  line-height: 40px;
  color: black;
  margin: 0;
}

.c0 form {
  font-weight: bold;
  font-size: 12px;
  line-height: 24px;
  color: black;
}

.c0 form input[type='checkbox'] {
  vertical-align: middle;
  margin-right: 10px;
  cursor: pointer;
}

.c0 form input[type='checkbox']:before {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: white;
}

.c0 form input[type='checkbox']:checked:after {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: black;
}

.c0 form label {
  display: block;
  margin-top: -24px;
  margin-left: 20px;
}

<div
  className="c0"
>
  <h5>
    Apply Tags
  </h5>
  <h6>
    Design Areas
  </h6>
  <form>
    <div>
      <input
        id="Art + Design"
        type="checkbox"
      />
      <label
        htmlFor="Art + Design"
      >
        Art + Design
      </label>
    </div>
    <div>
      <input
        id="Digital Design"
        type="checkbox"
      />
      <label
        htmlFor="Digital Design"
      >
        Digital Design
      </label>
    </div>
    <div>
      <input
        id="Industrial Design"
        type="checkbox"
      />
      <label
        htmlFor="Industrial Design"
      >
        Industrial Design
      </label>
    </div>
    <div>
      <input
        id="Policy Design"
        type="checkbox"
      />
      <label
        htmlFor="Policy Design"
      >
        Policy Design
      </label>
    </div>
    <div>
      <input
        id="Spatial Design"
        type="checkbox"
      />
      <label
        htmlFor="Spatial Design"
      >
        Spatial Design
      </label>
    </div>
  </form>
  <h6>
    Schools
  </h6>
  <form>
    <div>
      <input
        id="AAP"
        type="checkbox"
      />
      <label
        htmlFor="AAP"
      >
        AAP
      </label>
    </div>
    <div>
      <input
        id="Agriculture and Life Sciences"
        type="checkbox"
      />
      <label
        htmlFor="Agriculture and Life Sciences"
      >
        Agriculture and Life Sciences
      </label>
    </div>
    <div>
      <input
        id="Arts and Sciences"
        type="checkbox"
      />
      <label
        htmlFor="Arts and Sciences"
      >
        Arts and Sciences
      </label>
    </div>
    <div>
      <input
        id="Engineering"
        type="checkbox"
      />
      <label
        htmlFor="Engineering"
      >
        Engineering
      </label>
    </div>
    <div>
      <input
        id="Human Ecology"
        type="checkbox"
      />
      <label
        htmlFor="Human Ecology"
      >
        Human Ecology
      </label>
    </div>
    <div>
      <input
        id="ILR"
        type="checkbox"
      />
      <label
        htmlFor="ILR"
      >
        ILR
      </label>
    </div>
    <div>
      <input
        id="SC Johnson School of Business"
        type="checkbox"
      />
      <label
        htmlFor="SC Johnson School of Business"
      >
        SC Johnson School of Business
      </label>
    </div>
  </form>
</div>
`;
